spec/GraphQLOverHTTP.md

```diff
@@ -511,19 +511,16 @@ execution regardless of validation errors.
 
 ## Status Codes
 
-In case of errors that completely prevent the generation of a well-formed
-_GraphQL response_, the server SHOULD respond with the appropriate status code
-depending on the concrete error condition, and MUST NOT respond with a `2xx`
-status code when using the `application/graphql-response+json` media type.
+The status codes depends on the media type with which the GraphQL response will
+be served.
 
-Note: Typically the appropriate status code will be `400` (Bad Request).
-
-Note: This rule is "should" to maintain compatibility with legacy servers which
-can return 200 status codes even when this type of error occurs, but only when
-not using the `application/graphql-response+json` media type.
-
-Otherwise, the status codes depends on the media type with which the GraphQL
-response will be served:
+Note: when using the `application/graphql-response+json` media type, the status
+code is primarily aimed for intermediary servers that do not understand GraphQL.
+In these cases, clients can rely on the response being a well-formed _GraphQL
+response_ regardless of the status code. When using the `application/json` media
+type on the other hand, clients may use a `2xx` status code as an indication
+that the body contains a well-formed _GraphQL response_. See
+[processing a response](#sec-Processing-a-response) for more details.
```

Comment:
This paragraph mostly duplicates `6.5 processing a response` but is added for discoverability and also expliciting the role of the status code.